src/main/kotlin/org/opensearch/replication/metadata/store/ReplicationMetadataStore.kt

```diff
@@ -256,4 +270,15 @@ class ReplicationMetadataStore constructor(val client: Client, val clusterServic
 
     override fun doClose() {
     }
+
+    private suspend fun checkAndWaitForStoreHealth() {
```

Comment:
how is this waiting for store health ? From the code, I see that we are just making one successful call to cluster health and then asserting .

Reply:
Currently, the check is to ensure that the store index is up and can take the requests. If the call is successful, it can surface the information. Do you have any other checks in mind?

Reply:
sorry , I missed the `waitForYellowStatus` which would do the wait. So we are good here.